site/source/docs/compiling/Deploying-Pages.rst

```diff
@@ -0,0 +1,162 @@
+
+Additional build output files can also exist, depending on which features are used. If the Emscripten file packager is used, a binary ``out.data`` package is generated, along with an associated ``out.data.js`` loader file. Also Emscripten pthreads and Fetch APIs have their own associated Web Worker related script ``.js`` output files.
+
+Developers can choose to output either to JavaScript or HTML. If outputting JavaScript (``emcc -o out.js``), developer is expected to manually create the ``out.html`` main page in which the code is run in browsers. When targeting HTML with ``emcc -o out.html``, Emscripten will generate the HTML shell file automatically. This shell file can be customized by using the ``emcc -o out.html --shell-file path/to/custom_shell.html`` linker directive. Copy the `default minimal HTML shell file <https://github.com/kripken/emscripten/blob/master/src/shell_minimal.html>`_ from Emscripten repository to your project tree to get a good starting template for a customized shell file.
+
+The following sections offer tips for improving the site experience. 
+
+Optimizing Download Sizes
+=========================
+
+The biggest slowdown to speedy page loading is most often the need to download large amounts of asset data for the project, especially if the page is heavy on WebGL textures or geometry. Compiled code generally takes up more space than handwritten JavaScript, but machine code compresses efficiently. Therefore when hosting asm.js and WebAssembly, it is critical to ensure that all content is transferred using gzip compression, which all browsers and CDNs nowadays support built-in. Gzip compressing ``.wasm`` files give on average 60-75% size reductions compared to uncompressed ones, so it practically never makes sense to serve uncompressed files.
+
+- To serve gzip-compressed assets on a CDN, use a gzip compression tool and precompress asset files offline before uploading to the CDN. Some web servers support compressing files on the fly, but for static asset content, that should be avoided since it can be costly for the server CPU to keep recompressing the files. Adjust the configuration of the web server to host the precompressed files with the HTTP response header ``Content-Encoding: gzip``. This instructs web browsers that the downloaded content should be transparently uncompressed before handing the data off to the page itself.
+
+- WebAssembly has now shipped in `Firefox 52 and Chrome 57 <http://caniuse.com/#feat=wasm>`_. Emscripten also supports targeting WebAssembly, by using the ``-s BINARYEN=1`` linker flag. WebAssembly is an evolution of asm.js, and if your project already successfully compiles to asm.js, it is likely to already work with WebAssembly as well. Compressed WebAssembly output files are on average 50-60% smaller than compressed asm.js files, so the benefit is large.
```

Comment:
The data points I used as reference here were from Zen Garden: .asm.js.gz: 19316250 bytes,  .wasm.gz: 9660956 bytes. (-50.0%) and another partner demo that is not public. Now that I look closer, I realize these were from builds that include debugging symbols for profiling information. For Unity Ski Safari build which does not have debugging symbols, the comparison is asm.js.gz: 5139023 bytes, wasm.gz: 4024425 bytes (-21.7%), which is quite a bit more modest. 

I'll write `"Compressed WebAssembly output files can be around 20% smaller than compressed asm.js files, but for builds with debugging and profiling information, the difference can even be up to 50%, so the benefit is large."`, how does that read?